release-notes/rascal-0-40-x-release-notes.md

```diff
@@ -0,0 +1,92 @@
+---
+authors: [jvinju]
+title: "Rascal 0.40.x release notes"
+sidebar_position: 96
+---
+
+In this post we report on the Rascal release 0.40.x
+
+<!--truncate-->
+
+## Release 0.40.14 - November 5, 2024
+
+The public release 0.40.x follows release 0.28.x; many improvements have been made in projects that depends on the rascal interpreter and the standard library (the type checker, the VScode extensions, clair, etc.) Some of these improvements depend directly on fixes in the interpreter and additions to the standard library. 
+
+The support for XML, JSON and HTML as exchange formats has been improved or completely rewritten. The main feature that was added was the optional `loc src` keyword field that provides the exact location of each node as it appears in the XML, HTML or JSON source text.
+
+A new `mvn://` scheme was added to be able to address with brief but unique notation the jar files in the local M2 repository. It supports two modes:
+* `mvn://groupId!artifactId!optionAlComplexVersionString/pathInsideJar`, shorthand for identifying artifacts inside the mvn repository on this machine.
+* `mvn:///path/inside/m2/repository/jarFile.jar` for exploring the mvn repository from its root.
+If `maven.repo.local` is set, then that is the root of the M2 repository. Otherwise if `~/m2/.repository` exists (in the users ome directory) then that is used. Otherwise `mvn` is executed with `-Dexpression=settings.localRepository` to extract the setting from the current `pom.xml` file. This is _not_ the official order as `mvn` resolves these configuration options; but it was chosen for the sake of efficiency.
+
+A great deal of tests were fixed, enhanced or extended as a side-effect of the compiler project. Another visible aspect of the progress of the compiler is that now all Rascal runtime values (from vallang and beyond) now support `getFingerprint()` methods which help in optimizing patter matching and dispatch in generated code by the compiler. These  methods' return values have become a strict contract for future implementations of Rascal values, including parse trees and reified types and first-class functions.
+
+The following issues were solved: 
+* string `visit` with unicode characters had a bug
+* `util::ShellExec` had some IO synchronization issues which were resolved.
+* The JUnit test runners report exceptions better now.
+* The broken and unused `stdout://` and `stdin://` resolvers were removed.
+* The `IO::watch` functionality was improved for logical source locations. However the system is still too slow on Mac to work effectively in an interactive development environment.
+* TODO
```

Comment:
maybe also include at least one line about various the performance fixes?

Reply:
Just in general? I don't have the specifics ready